Tests/SubprocessTests/IntegrationTests.swift

```
@@ -0,0 +1,2296 @@
+import _SubprocessCShims
+@testable import Subprocess
+
+@Suite("Subprocess Integration (End to End) Tests", .serialized)
+struct SubprocessIntegrationTests {}
+
+// MARK: - Executable Tests
+extension SubprocessIntegrationTests {
+    @Test func testExecutableNamed() async throws {
+        let message = "Hello, world!"
+        #if os(Windows)
+        let setup = TestSetup(
+            executable: .name("cmd.exe"),
+            arguments: .init(["/c", "echo", message])
+        )
+        #else
+        let setup = TestSetup(
+            executable: .name("echo"),
+            arguments: .init([message])
+        )
+        #endif
+
+        // Simple test to make sure we can find a common utility
+        let result = try await _run(
+            setup,
+            input: .none,
+            output: .string(limit: 32),
+            error: .discarded
+        )
+        #expect(result.terminationStatus.isSuccess)
+        // rdar://138670128
+        // https://github.com/swiftlang/swift/issues/77235
+        let output = result.standardOutput?
+            .trimmingNewLineAndQuotes()
+        // Windows echo includes quotes
+        #expect(output == message)
+    }
+
+    @Test func testExecutableNamedCannotResolve() async {
+        #if os(Windows)
+        let expectedError = SubprocessError(
+            code: .init(.executableNotFound("do-not-exist")),
+            underlyingError: .init(rawValue: DWORD(ERROR_FILE_NOT_FOUND))
+        )
+        #else
+        let expectedError = SubprocessError(
+            code: .init(.executableNotFound("do-not-exist")),
+            underlyingError: .init(rawValue: ENOENT)
+        )
+        #endif
+
+        await #expect(throws: expectedError) {
+            _ = try await Subprocess.run(.name("do-not-exist"), output: .discarded)
+        }
+    }
+
+    @Test func testExecutableAtPath() async throws {
+        #if os(Windows)
+        let cmdExe = ProcessInfo.processInfo.environment["COMSPEC"] ??
```

needs to be treated case insensitively (it's ComSpec by default), and you might as well make this #require without falling back to a hardcoded path, or just call GetSystemDirectoryW and join cmd.exe to that.